jac/jaclang/compiler/program.py

```diff
@@ -66,6 +67,13 @@ def __init__(self, main_mod: Optional[uni.ProgramModule] = None) -> None:
         self.py_raise_map: dict[str, str] = {}
         self.errors_had: list[Alert] = []
         self.warnings_had: list[Alert] = []
+        self.type_evaluator: TypeEvaluator | None = None
+
+    def get_type_evaluator(self) -> TypeEvaluator:
+        """Return the type evaluator."""
+        if not self.type_evaluator:
+            self.type_evaluator = TypeEvaluator(program=self)
+        return self.type_evaluator
```

Comment:
Since the type evaluator will load the builtin module and prefetch some symbols, we don't necessarily have to do that every time we create an instance, instead we lazy load and only run when type check if requested